### PR TITLE
[New tool] XSD to BigQuery schema generator

### PR DESCRIPTION
**Pull request by aymanfarhat**
Hello! I'd like to add a new tool I've been working on and using related to generating BigQuery schemas to represent deeply nested and repeated XML content with custom types. 

Features:

- Translates custom and native XSD types to the nearest closest datatypes available in BigQuery
- Supports nested XML/XSD element structure regardless of depth level represented as STRUCT
- Supports repeated fields in XML files, represented ad REPEATED in BigQuery
- Generates a JSON schema that can be directly used in the creation of BigQuery tables

In terms of code:

- Lint checks and style guide based on Google's open source style guide linked to in this repo enforced via pylint + yapf used for formatting
- Unit tests added, mainly to test the functionality and corner cases of the main functions

Thanks in advance for taking the time to review! :)

### CODE REVIEW

File: tools/xsd-to-bigquery-schema/.pylintrc [hunk 1]
```diff
@@ -0,0 +1,441 @@
+# This Pylint rcfile contains a best-effort configuration to uphold the
```
Comment (iht) [inline tools/xsd-to-bigquery-schema/.pylintrc:1]: Do we need this file? This repo has already a Python linter with a predefined configuration. This file seems to comply with the Google Python style guide, but still, I would remove it and rely on the configuration provided by the repo.

File: tools/xsd-to-bigquery-schema/tests/sample_data/customer_orders.xsd [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (iht) [inline tools/xsd-to-bigquery-schema/tests/sample_data/customer_orders.xsd:1]: Copyright header

**Review by iht — CHANGES_REQUESTED** [review]
Please see comment about a missing copyright header in one of the files. Other than that, this LGTM.

Also, I am not sure that we need the `.pylintrc` file, given that the repo already checks Python code for style issues (and probably using the same rules encoded in that file).

File: tools/xsd-to-bigquery-schema/.pylintrc (hunk 1 above)
Comment (aymanfarhat) [reply tools/xsd-to-bigquery-schema/.pylintrc]: Yes I agree, it won't be needed, got carried with the code since I was working on this separately then moved to the PSO repo. Just removed it now.

File: tools/xsd-to-bigquery-schema/tests/sample_data/customer_orders.xsd (hunk 2 above)
Comment (aymanfarhat) [reply tools/xsd-to-bigquery-schema/tests/sample_data/customer_orders.xsd:16]: Thanks for the catch, updated.

**Review by iht — APPROVED** [review]
LGTM
